### PR TITLE
[DEV-10630] Autocomplete recipient partial search 

### PR DESCRIPTION
**Pull request by sukhusaspending**
**Description:**
When you enter a partial query, such as ‘YV4GH5P,’ for the Unique Entity Identifier (UEI) ‘YV4GH5PF28Y9,’ the search cannot be completed without the full UEI name. However, the designs for Award Search 2.0 include a feature that allows users to start typing a UEI and receive autocomplete suggestions, similar to how recipient names are handled.

**Technical details:**
search query was updated to add * (like) operator to input values 

**Requirements for PR merge:**

1. [X] Unit & integration tests updated
2. [X] API documentation updated
3. [X] Necessary PR reviewers:
    - [X] Backend
8. [X] Jira Ticket [DEV-10630](https://federal-spending-transparency.atlassian.net/browse/DEV-10630):
    - [X] Link to this Pull-Request
   

**Area for explaining above N/A when needed:**
```
```

### CODE REVIEW

**Comment by sukhusaspending** [comment]
Please review the pull request

**Comment by sukhusaspending** [comment]
travis build is completed and tests are passing now. pls review the pull request

File: usaspending_api/references/v2/views/recipients.py [hunk 1]
```diff
@@ -87,16 +87,19 @@ def _create_es_search(self, search_text: str, recipient_levels: List[str], limit
...
+        # query = ES_Q(
+        #     "bool",
+        #     should=[
+        #         ES_Q("query_string", query=f"*{search_text}*", fields=es_recipient_search_fields),
+        #         ES_Q("match", recipient_name=search_text),
+        #         ES_Q("match", uei=search_text),
+        #         ES_Q("match", duns=search_text),
+        #     ],
+        #     minimum_should_match=1,
+        # )
```
Comment (aguest-kc) [inline usaspending_api/references/v2/views/recipients.py:99]: Can you remove these lines since they've been commented out and replaced by lines 100-102?

**Review by aguest-kc — CHANGES_REQUESTED** [review]

File: usaspending_api/references/v2/views/recipients.py (hunk 1 above)
Comment (sukhusaspending) [reply usaspending_api/references/v2/views/recipients.py]: done. Pls review again

**Review by aguest-kc — APPROVED** [review]

**Pull request merged**